presto-main-base/src/main/java/com/facebook/presto/execution/scheduler/nodeSelection/SimpleNodeSelector.java

```diff
@@ -179,7 +193,14 @@ public SplitPlacementResult computeAssignments(Set<Split> splits, List<RemoteTas
             }
 
             SplitWeight splitWeight = split.getSplitWeight();
-            Optional<InternalNodeInfo> chosenNodeInfo = chooseLeastBusyNode(splitWeight, candidateNodes, assignmentStats::getTotalSplitsWeight, preferredNodeCount, maxSplitsWeightPerNode, assignmentStats);
+            Optional<InternalNodeInfo> chosenNodeInfo = Optional.empty();
+
+            if (taskLoadSplitWeightProvider != null) {
+                chosenNodeInfo = chooseLeastBusyNode(splitWeight, candidateNodes, taskLoadSplitWeightProvider, preferredNodeCount, maxSplitsWeightPerTask, assignmentStats);
+            }
+            else {
+                chosenNodeInfo = chooseLeastBusyNode(splitWeight, candidateNodes, assignmentStats::getTotalSplitsWeight, preferredNodeCount, maxSplitsWeightPerNode, assignmentStats);
+            }
             if (!chosenNodeInfo.isPresent()) {
```

Comment:
do we still want to fall back to this option for task based split scheduling or should this be inside the else block?

Reply:
@rschlussel 
I honestly don't quite understand this fallback, so decided to keep it for both node- and task-based scheduling, in case it is critical.

Do you have an idea why we have it for node-based scheduling?

I can investigate code more in order to understand this, but not sure in the result. :)